quesma/queryparser/pancake_json_rendering.go

```diff
@@ -106,13 +120,64 @@ func (p *pancakeJSONRenderer) potentiallyRemoveExtraBucket(layer *pancakeModelLa
 	return bucketRows, subAggrRows
 }
 
-func (p *pancakeJSONRenderer) layerToJSON(layerIdx int, layers []*pancakeModelLayer, rows []model.QueryResultRow) (model.JsonMap, error) {
+func (p *pancakeJSONRenderer) combinatorBucketToJSON(remainingLayers []*pancakeModelLayer, rows []model.QueryResultRow) (model.JsonMap, error) {
+	layer := remainingLayers[0]
+	switch queryType := layer.nextBucketAggregation.queryType.(type) {
+	case bucket_aggregations.SamplerInterface, bucket_aggregations.FilterAgg:
+		selectedRows := p.selectMetricRows(layer.nextBucketAggregation.InternalNameForCount(), rows)
+		aggJson := layer.nextBucketAggregation.queryType.TranslateSqlResponseToJson(selectedRows, 0)
+		subAggr, err := p.layerToJSON(remainingLayers[1:], rows)
+		if err != nil {
+			return nil, err
+		}
+		return util.MergeMaps(context.Background(), aggJson, subAggr, model.KeyAddedByQuesma), nil
```

Comment:
I think we should propagate our `context` here instead of creating new with `context.Background()`